spec/chainable_queries_spec.rb

```diff
@@ -129,6 +129,38 @@ def invalid?
         expect(subject.find_by(foo: {gte: 123})).to eq subject
         expect(subject.query.parameters).to include('fields.foo[gte]' => 123)
       end
+
+      it 'when multiple fields' do
+        expect(subject.find_by(foo: true, bar: 123)).to eq subject
+        expect(subject.query.parameters).to include('fields.foo' => true , 'fields.bar' => 123)
+      end
+
+      it 'supports sys fields' do
+        expect(subject.find_by('sys.id': 'foo')).to eq subject
+        expect(subject.query.parameters).to include('sys.id' => 'foo')
+      end
+    end
+
+    describe '::paginate' do
+      it 'defaults to first page and 100 items' do
+        expect(subject.paginate).to eq subject
+        expect(subject.query.parameters).to include('limit' => 100, 'skip' => 0)
+      end
+
+      it 'requesting second page will add page_size to skip' do
+        expect(subject.paginate(2)).to eq subject
+        expect(subject.query.parameters).to include('limit' => 100, 'skip' => 100)
+      end
+
+      it 'can change page_size' do
+        expect(subject.paginate(1, 20)).to eq subject
+        expect(subject.query.parameters).to include('limit' => 20, 'skip' => 0)
+      end
+
+      it 'can change page_size and select a different page' do
+        expect(subject.paginate(3, 20)).to eq subject
+        expect(subject.query.parameters).to include('limit' => 20, 'skip' => 40)
+      end
```

Comment:
Do you mind checking the outliers too? Like `.paginate(-1,0)` or `.paginate('three')`, and so on?

Reply:
👌 will add.